src/SFML/Network/Http.cpp

```diff
@@ -154,8 +155,7 @@ m_minorVersion(0)
 ////////////////////////////////////////////////////////////
 const std::string& Http::Response::getField(const std::string& field) const
 {
-    FieldTable::const_iterator it = m_fields.find(toLower(field));
-    if (it != m_fields.end())
+    if (auto it = m_fields.find(toLower(field)); it != m_fields.end())
```

Comment:
I like the if-with-initializer, I would like to encourage its use. It might not be that useful here, but it's nice to start to be consistent with the feature. I will remove the `else` branch (redundant) as well, which does limit the scope of `it` in a useful way.

Reply:
I'm not really convinced that:
```cpp
if (auto it = m_fields.find(toLower(field)); it != m_fields.end())
{
     return it->second;
}

static const std::string empty = "";
return empty;
```
is more readable than:
```cpp
auto it = m_fields.find(toLower(field));
if (it != m_fields.end())
{
      return it->second;
}
else
{
     static const std::string empty = "";
     return empty;
}
```
or better, without negation and useless static:
```cpp
auto it = m_fields.find(toLower(field));
if (it == m_fields.end())
     return std::string();
else
     return it->second;
```

In the latter example, it's immediately clear that:
* the condition is to check an iterator against `end()`
* there are two outcomes (if/else) on which it is dispatched

Any other view on this, e.g. @eXpl0it3r? Could of course be more compact with `{}` and `?:`, but this is not necessarily what we're optimizing for 😉 

--- 

Apart from that, we should just return a default-constructed `std::string` instance, a reasonable implementation should not allocate memory.

Reply:
I guess readability for a new language/syntax feature is a bit tricky, as the good-old style of course will be "more readable" as we've been used to it for the past decade or longer.
I like the functionality that if-with-initializer provides, so I would at least try it out for a while before judging its readability.

I'm very much used to early-returns and dropping useless conditional branching (i.e. no `else` block after `return` in an `if` block). So that part would be fine for me.

Is the `static const std::string` really "useless"? As it would ensure that it's only created once, or am I wrong?

One alternative not yet mentioned would be:
```
auto  it = m_fields.find(toLower(field));
return it == m_fields.end() ? std::string() : it->second;
```

I have no strong opinions on any of the styles and don't know how much time we should invest in finding the most pleasing solution. 😄

Reply:
I don't think we should focus on the style of this code much. All mentioned alternatives are quite readable. I prefer my own because it limits the scope of the iterator to the `if`, and reducing scope of variables is IMHO a good thing.

Also, we are returning `const std::string&', so we cannot return a temporary.

Reply:
Regarding if-with-initializer: it can make sense if we have a bigger scope and we want to keep lifetime of variables as local as possible (this may also be an indicator for a separate method). However, none of the advantages play out here:
* it's not a big scope
* there is no code coming after, which could accidentally use the variable
* it's not an early return, it's the last statement in the function

I agree it's not worth discussing for hours, but maybe precisely for that reason we should address the larger issues before getting into bikeshedding territory 🙂  we can leave this particular change in, but please let's not hunt down `if` statements in SFML now 😆 

---

> Also, we are returning const std::string&, so we cannot return a temporary. 
> We are also not allocating any memory by creating an empty static std::string.

True, I missed that we return a const-reference.

We're not dynamically allocating memory, but we reserve static memory until the end of the program, even if this method is only used once. Plus, we add potential thread unsafety. Both are very minor issues here, but keep in mind that they're trade-offs for another minor issue (copying a `std::string` in a code that likely suffers from much larger performance impact, network latency). 

Anyway, this just as an aside 🙂